lib/api/repository/oauth_repository.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter_survey/api/oauth_service.dart';
+import 'package:flutter_survey/api/request/oauth_token_request.dart';
+import 'package:flutter_survey/api/response/oauth_token_response.dart';
+import 'package:flutter_survey/env.dart';
+import 'package:injectable/injectable.dart';
+
+abstract class OauthRepository {
+  Future<OauthTokenResponse> login({
+    required String email,
+    required String password,
+  });
+}
+
+@Singleton(as: OauthRepository)
+class OauthRepositoryImpl extends OauthRepository {
+  late OauthService _oauthService;
+
+  OauthRepositoryImpl(this._oauthService);
+
+  @override
+  Future<OauthTokenResponse> login({
+    required String email,
+    required String password,
+  }) {
+    return _oauthService
+        .login(OauthTokenRequest(
+          email: email,
+          password: password,
+          clientId: Env.basicAuthClientId,
+          clientSecret: Env.basicAuthClientSecret,
+        ))
+        .then((response) => response.data.attributes);
```

Comment:
I'd recommend to also wrap the API error from here, because otherwise the UseCase will have to convert the error using `NetworkExceptions.fromDio(exception)` - in that case, we are allowing UseCase to know the existence of `Dio` - so better throw the NetworkExceptions from here, we can then catch the exception of our own from UseCase

Reply:
Fixed https://github.com/luongvo/flutter-survey/pull/52/commits/12591c59f6d1af872d98360299ce80b70f9e0f89